spec.bs

```diff
@@ -8372,49 +8786,81 @@ To <dfn>batch and fetch trusted scoring signals</dfn> given a [=trusted scoring
+        combine requests. All entries need to be handled eventually.
+
+        1. [=list/Remove=] |entriesToBatch| from |batcher|'s [=trusted scoring signals batcher/
+          request map=][|key|].
+        1. If |batcher|'s [=trusted scoring signals batcher/request map=][|key|] [=list/is empty=],
+          then [=map/remove=] |key| from |batcher|'s [=trusted scoring signals batcher/request map=].
+        1. Let |fullSignalsUrl| be the result of [=building batched trusted scoring signals url=] for
+          |entriesToBatch|.
+        1. Let |seller| be |entriesToBatch|[0]'s [=trusted scoring signals request/seller=].
+        1. Let |scriptFetcher| be |entriesToBatch|[0]'s [=trusted scoring signals request/seller
+          script fetcher=].
+        1. Let |policyContainer| be |entriesToBatch|[0]'s [=trusted scoring signals request/policy
+          container=].
+        1. If |fullSignalsUrl|'s [=url/origin=] is not [=same origin=] with |seller| then:
+          1. Let |allowCrossOriginTrustedScoringSignalsFrom| be the result of [=wait for cross origin
+            trusted scoring signals authorization from a fetcher=] given |scriptFetcher|.
+          1. If |allowCrossOriginTrustedScoringSignalsFrom| does not [=list/contain=]
+            |fullSignalsUrl|'s [=url/origin=], set |fullSignalsUrl| to null.
+        1. Let |result| be failure.
+        1. If |fullSignalsUrl| is not null:
+          1. Let «|allTrustedScoringSignals|, <var ignore>ignored</var>, |scoringDataVersion|» be
+            the result of [=fetching trusted signals=] with |fullSignalsUrl|, |seller|,
+            |policyContainer|, and false.
+          1. If |allTrustedScoringSignals| is an [=ordered map=]:
+            1. Set |result| to a new [=trusted scoring signals reply=]
+            1. Set |result|'s [=trusted scoring signals reply/all trusted scoring signals=] to
+              |allTrustedScoringSignals|.
+            1. Set |result|'s [=trusted scoring signals reply/data version=] to |scoringDataVersion|.
+        1. [=list/For each=] |entry| in |entriesToBatch|:
+          1. Set |entry|'s [=trusted scoring signals request/reply=] to |result|.
+    1. Otherwise:
+      1. Set |entriesToBatch| to |batcher|'s [=trusted scoring signals batcher/request map=][|key|].
+      1. [=map/remove=] |key| from |batcher|'s [=trusted scoring signals batcher/request map=].
+      1. Let « |requestBody|, |renderUrlIdMap|, |context| » be the result of [=building batched
+        trusted key value scoring signals request body=] for |entriesToBatch|.
+      1. Let |baseUrl| be |entriesToBatch|[0]'s [=trusted scoring signals request/base url=].
       1. Let |seller| be |entriesToBatch|[0]'s [=trusted scoring signals request/seller=].
       1. Let |scriptFetcher| be |entriesToBatch|[0]'s [=trusted scoring signals request/seller
         script fetcher=].
       1. Let |policyContainer| be |entriesToBatch|[0]'s [=trusted scoring signals request/policy

```

Comment:
Nit: these two variables seem to be only used once, so better to just inline them, instead of adding extra steps and variables.